### PR TITLE
Update options for guidelines

### PR DESCRIPTION
**Pull request by rickle-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java [hunk 1]
```diff
@@ -193,32 +194,31 @@ public void uploadFromFile(String filePath, ParallelTransferOptions parallelTran
...
-                                                              Duration timeout, Context context)
+                Context context)
         throws UncheckedIOException {
         Mono<Response<BlockBlobItem>> upload =
             this.encryptedBlobAsyncClient.uploadFromFileWithResponse(options)
                 .subscriberContext(FluxUtil.toReactorContext(context));
 
         try {
-            return StorageImplUtils.blockWithOptionalTimeout(upload, timeout);
+            return StorageImplUtils.blockWithOptionalTimeout(upload, options.getTimeout());
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java:221]: I think in some APIs Duration is outside the options and in some it is in the options. We should probably figure out a consistent way to do it

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java [hunk 2]
```diff
@@ -278,23 +280,28 @@
...
+    public Mono<Response<BlockBlobItem>> uploadWithResponse(BlobParallelUploadOptions options) {
         try {
-            BlobParallelUploadOptions finalOptions = options == null ? new BlobParallelUploadOptions() : options;
-            final Map<String, String> metadataFinal = finalOptions.getMetadata() == null
-                ? new HashMap<>() : finalOptions.getMetadata();
-            finalOptions.setMetadata(metadataFinal);
+            Objects.requireNonNull(options);
+            final Map<String, String> metadataFinal = options.getMetadata() == null
+                ? new HashMap<>() : options.getMetadata();
+            options.setMetadata(metadataFinal);
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:296]: Some say that mutating input parameters isn't good practice. 

https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern

looks like we do a copy few lines below anyway.

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java (hunk 1 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java:221]: Any method that has been given an overload which accepts options has the timeout moved to the options. Any api where the timeout is still its own parameter has not yet had an options overload added. (Unless I've missed something)

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java [hunk 3]
```diff
@@ -278,23 +280,28 @@
...
      */
     @Override
-    public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data, BlobParallelUploadOptions options) {
+    public Mono<Response<BlockBlobItem>> uploadWithResponse(BlobParallelUploadOptions options) {
         try {
-            BlobParallelUploadOptions finalOptions = options == null ? new BlobParallelUploadOptions() : options;
-            final Map<String, String> metadataFinal = finalOptions.getMetadata() == null
-                ? new HashMap<>() : finalOptions.getMetadata();
-            finalOptions.setMetadata(metadataFinal);
+            Objects.requireNonNull(options);
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:293]: Should we use our "utils" for that?

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java (hunk 2 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:296]: I don't have to set it on the options. I can use a local variable if that's better?

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java (hunk 3 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:293]: Yes. I thought I went through and changed all these, but I guess I missed some. Nice catch!

File: sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java (hunk 2 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:296]: sounds good,

Comment (kasobol-msft) [reply sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java:296]: btw. Collections.emtpyMap() might be more optimal.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobQueryOptions.java [hunk 4]
```diff
@@ -32,6 +34,19 @@
     public BlobQueryOptions(String expression) {
         StorageImplUtils.assertNotNull("expression", expression);
         this.expression = expression;
+        this.outputStream = null;
+    }
+
+    /**
+     * Constructs a {@link BlobQueryOptions}.
+     * @param expression The query expression.
+     * @param outputStream Gets the OutputStream where the downloaded data will be written.
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobQueryOptions.java:43]: nit: this should just be "the outputstream..."

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java [hunk 5]
```diff
@@ -161,10 +161,9 @@ public AppendBlobItem create(boolean overwrite) {
      * @param context Additional context that is passed through the Http pipeline during the service call.
      * @return A {@link Response} whose {@link Response#getValue() value} contains the created appended blob.
      */
-    public Response<AppendBlobItem> createWithResponse(AppendBlobCreateOptions options, Duration timeout,
-        Context context) {
+    public Response<AppendBlobItem> createWithResponse(AppendBlobCreateOptions options, Duration timeout, Context context) {
         return StorageImplUtils.blockWithOptionalTimeout(appendBlobAsyncClient.
-            createWithResponse(options, context), timeout);
+            createWithResponse(options, context), options == null ? null : timeout);
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java:166]: This should just be timeout

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java [hunk 6]
```diff
@@ -305,13 +301,10 @@ public BlockBlobItem upload(InputStream data, long length, boolean overwrite) {
...
-    public Response<BlockBlobItem> uploadWithResponse(InputStream data, long length,
-        BlockBlobSimpleUploadOptions options, Duration timeout, Context context) {
-        Objects.requireNonNull(data);
-        Flux<ByteBuffer> fbb = Utility.convertStreamToByteBuffer(data, length,
-            BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE);
-        Mono<Response<BlockBlobItem>> upload = client
-            .uploadWithResponse(fbb.subscribeOn(Schedulers.elastic()), length, options, context);
+    public Response<BlockBlobItem> uploadWithResponse(BlockBlobSimpleUploadOptions options, Duration timeout,
+        Context context) {
+        Objects.requireNonNull(options);
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java:306]: Should this be StorageImplUtils?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceAPITest.groovy [hunk 7]
```diff
@@ -355,13 +355,13 @@ class ServiceAPITest extends APISpec {
...
-                new BlobParallelUploadOptions().setTags(tags), null, null)
+            cc.getBlobClient(generateBlobName()).uploadWithResponse(
+                new BlobParallelUploadOptions(defaultInputStream.get(), defaultDataSize).setTags(tags), null, null)
         }
 
         when: "Consume results by page"
-        primaryBlobServiceClient.findBlobsByTags("\"tag\"='value'",
-            new FindBlobsOptions().setMaxResultsPerPage(PAGE_RESULTS), Duration.ofSeconds(10))
+        primaryBlobServiceClient.findBlobsByTags(new FindBlobsOptions("\"tag\"='value'")
+            .setMaxResultsPerPage(PAGE_RESULTS).setTimeout(Duration.ofSeconds(10)))
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceAPITest.groovy:364]: This shouldnt be possible anymore right?

**Pull request merged**